src/main/java/com/google/javascript/gents/GentsCodeGenerator.java

```diff
@@ -54,11 +56,20 @@ protected void add(Node n, Context ctx) {
         break;
     }
 
-    // Add an empty line if the next node is further away than the next line.
+    addNewlines(n);
+  }
+
+  /** Add newlines to the generated source */
+  private void addNewlines(Node n) {
     Node nextNode = n.getNext();
     if (nextNode != null) {
       int lineSpacing = nextNode.getLineno() - n.getLineno();
-      if (lineSpacing > 1) {
+
+      // Add an empty line if:
+      //   - the next node is further away than the next line
+      //   - the next node is a class function (but the current node is not a comment)
+      if (lineSpacing > 1 ||
+          (nextNode.getToken() == Token.MEMBER_FUNCTION_DEF ^ n.getToken() == Token.EMPTY)) {
```

Comment:
One of the newline conditions here is "current node is empty and next token is not a member function" - i.e. add an newline after all empty tokens except those followed by a member function. That seems non-intuitive to me; why is it what we want?